internal/controller/operatorconfigmap_controller.go

```diff
@@ -810,84 +796,74 @@ func (c *OperatorConfigMapReconciler) reconcileNoobaaOperatorSubscription() erro
-		}
-	}
+	c.log.Info("successfully reconcile webhook service")
 	return nil
 }
 
-func (c *OperatorConfigMapReconciler) reconcileCephCSIOperatorSubscription() error {
-	cephCsiOperatorSubscription, err := c.getSubscriptionByPackageName("cephcsi-operator")
-	if kerrors.IsNotFound(err) {
-		cephCsiOperatorSubscription, err = c.getSubscriptionByPackageName("ceph-csi-operator")
-	}
+func (c *OperatorConfigMapReconciler) reconcileSubscriptionLabelForWebhook() error {
+	sub, err := c.getSubscriptionByPackageName("ocs-client-operator")
 	if err != nil {
 		return err
 	}
-	if c.subscriptionChannel != "" && c.subscriptionChannel != cephCsiOperatorSubscription.Spec.Channel {
-		cephCsiOperatorSubscription.Spec.Channel = c.subscriptionChannel
-		if err := c.update(cephCsiOperatorSubscription); err != nil {
-			return fmt.Errorf("failed to update subscription channel of 'cephcsi-operator' to %v: %v", c.subscriptionChannel, err)
+	if utils.AddLabel(sub, subscriptionLabelKey, subscriptionLabelValue) {
+		if err := c.update(sub); err != nil {
+			return err
 		}
 	}
 	return nil
 }
 
-func (c *OperatorConfigMapReconciler) reconcileRecipeOperatorSubscription() error {
-	recipeOperatorSubscription, err := c.getSubscriptionByPackageName("recipe")
-	if err != nil {
-		return err
-	}
-	if c.subscriptionChannel != "" && c.subscriptionChannel != recipeOperatorSubscription.Spec.Channel {
-		recipeOperatorSubscription.Spec.Channel = c.subscriptionChannel
-		if err := c.update(recipeOperatorSubscription); err != nil {
-			return fmt.Errorf("failed to update subscription channel of 'recipe' to %v: %v", c.subscriptionChannel, err)
+func (c *OperatorConfigMapReconciler) reconcileSubscriptions() error {
+	installPlansForApproval := map[string]struct{}{}
+
+	for _, pkgName := range managedPackageNames {
+		sub, err := c.getSubscriptionByPackageName(pkgName)
+		if client.IgnoreNotFound(err) != nil {
+			return err
 		}
-	}
-	return nil
-}
+		if sub.UID != "" {
+			if c.subscriptionChannel != "" && c.subscriptionChannel != sub.Spec.Channel {
+				sub.Spec.Channel = c.subscriptionChannel
+				if err := c.update(sub); err != nil {
+					return fmt.Errorf("failed to update subscription channel of %s to %v: %v", sub.Name, c.subscriptionChannel, err)
+				}
+			}
 
-func (c *OperatorConfigMapReconciler) reconcileODFSnapshotterSubscription() error {
-	odfSnapshotterSubscription, err := c.getSubscriptionByPackageName("odf-external-snapshotter-operator")
-	if err != nil {
-		return err
-	}
-	if c.subscriptionChannel != "" && c.subscriptionChannel != odfSnapshotterSubscription.Spec.Channel {
-		odfSnapshotterSubscription.Spec.Channel = c.subscriptionChannel
-		if err := c.update(odfSnapshotterSubscription); err != nil {
-			return fmt.Errorf("failed to update subscription channel of 'odf-external-snapshotter-operator' to %v: %v", c.subscriptionChannel, err)
+			ipRef := sub.Status.InstallPlanRef
+			if ipRef != nil {
+				installPlansForApproval[ipRef.Name] = struct{}{}
+			}
```

Comment:
This does not make sense, approval of install planes has nothing to do with the subscriptions. The content of the install plan does not matter.